inc/snmpcredential.class.php

```diff
@@ -0,0 +1,170 @@
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with GLPI. If not, see <http://www.gnu.org/licenses/>.
+ * ---------------------------------------------------------------------
+ */
+
+if (!defined('GLPI_ROOT')) {
+   die("Sorry. You can't access this file directly");
+}
+
+use Glpi\Application\View\TemplateRenderer;
+
+/**
+ * SNMP credentials
+ */
+class SNMPCredential extends CommonDBTM {
+
+   // From CommonDBTM
+   public $dohistory                   = true;
+   static $rightname = 'computer';
+
+   static function getTypeName($nb = 0) {
+      return _n('SNMP credential', 'SNMP credentials', $nb);
+   }
+
+   public function rawSearchOptions() {
+      $tab = parent::rawSearchOptions();
+
+      $tab[] = [
+         'id'            => '2',
+         'table'         => $this->getTable(),
+         'field'         => 'community',
+         'name'          => __('Community'),
+         'datatype'      => 'string',
+         'massiveaction' => false,
+      ];
+
+      return $tab;
+   }
+
+   /**
+    * Define tabs to display on form page
+    *
+    * @param array $options
+    * @return array containing the tabs name
+    */
+   function defineTabs($options = []) {
+
+      $ong = [];
+      $this->addDefaultFormTab($ong);
+      $this->addStandardTab('Log', $ong, $options);
+
+      return $ong;
+   }
+
+   public function showForm($ID, $options = []) {
+      $this->initForm($ID, $options);
+      TemplateRenderer::getInstance()->display('components/form/snmpcredential.html.twig', [
+         'item'   => $this,
+         'params' => $options,
+      ]);
+
+      return true;
+   }
+
+   /**
+    * Real version of SNMP
+    *
+    * @return string
+    */
+   public function getRealVersion(): string {
+      switch ($this->fields['snmpversion']) {
+         case 1:
+         case 3:
+            return (string)$this->fields['snmpversion'];
+         case 2:
+            return '2c';
+         default:
+            return '';
+
+      }
+   }
+
+   /**
+    * Get SNMP authentication protocol
+    *
+    * @return string
+    */
+   function getAuthProtocol(): string {
+      switch ($this->fields['authentication']) {
+         case 1:
+            return 'MD5';
+         case 2:
+            return 'SHA';
+         default:
+            return '';
+      }
+      return '';
+   }
+
+   /**
+    * Get SNMP encryption protocol
+    *
+    * @return string
+    */
+   function getEncryption(): string {
+      switch ($this->fields['encryption']) {
+         case 1:
+            return 'DES';
+         case 2:
+            return 'AES';
+         case 5:
+            return '3DES';
+         default:
+            return '';
+      }
+   }
+
+   protected function prepareInputs(array $input): array {
+      if (isset($input['auth_passphrase'])) {
+         $input['auth_passphrase'] = Toolbox::sodiumEncrypt($input['auth_passphrase']);
```

Comment:
Note, encrypt/decrypt function will be deprecated soon.
This is just a reminder we should think about change theses line in #9729 after merge of the current pr